### PR TITLE
Thirst Mechanics

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Adds thirst mechanics. Thirst mechanics work exactly like hunger mechanics where if you don't drink enough liquid, you become thirsty and move slower. Several other mechanics involving drinks were also changed.

For organic races, you gain hydration by consuming beverages and water. Water is the most effective way of hydrating yourself. IPCs are not affected by hydration and do not have to worry about it. If you overeat or overdrink, your nutrition decreases faster until you are of normal hunger and thirst levels.

Penalties for not eating or drinking are now more severe. If you refuse to drink or eat all shift, you will be up to 33% slower and take twice as much pain damage. The amount scales based on your hunger or thirst levels. If you are considered hungry or thirsty, it starts scaling until you're starving or seriously dehydrated. If you overeat and overdrink as well, you get a ~5% reduced movement and pain tolerance with each mechanic.

The icons for hunger have been updated to be more specific, and they match the new thirst indicators as well. https://i.imgur.com/QQbWxYI.png

You can now also inject beverages into your blood for x4 effectiveness. Before, IV dripping yourself with vodka did nothing but give you a small amount of toxins, but now it should be a hell of a party.

Dionaea can now be affected by beverages with respect to nutrition gains and temperature changes. Dionaea cannot be affected by some effects, such as stuttering and intoxication, still.

Healing effects for certain beverages (tea, milk) were removed. This a HRP server, milk shouldn't heal you.

### CODE REVIEW

**Comment by BurgerLUA** [comment]
!wip

**Comment by BotBOREALIS** [comment]
Icons with diff:
<details><summary>icons/mob/screen/generic.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
nutrition0|nutrition0|nutrition0|Modified
nutrition1|nutrition1|nutrition1|Modified
nutrition2|nutrition2|nutrition2|Modified
nutrition3|nutrition3|nutrition3|Modified
nutrition4|nutrition4|nutrition4|Modified
nutrition5||nutrition5|Created
thirst5||thirst5|Created
thirst4||thirst4|Created
thirst3||thirst3|Created
thirst2||thirst2|Created
thirst1||thirst1|Created
thirst0||thirst0|Created
</details>
<details><summary>icons/mob/screen/midnight.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
nutrition0|nutrition0|nutrition0|Modified
nutrition2|nutrition2|nutrition2|Modified
nutrition3|nutrition3|nutrition3|Modified
nutrition4|nutrition4|nutrition4|Modified
nutrition5||nutrition5|Created
thirst5||thirst5|Created
thirst4||thirst4|Created
thirst3||thirst3|Created
thirst2||thirst2|Created
thirst1||thirst1|Created
thirst0||thirst0|Created
</details>
<details><summary>icons/mob/screen/old-noborder.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
nutrition0|nutrition0|nutrition0|Modified
nutrition2|nutrition2|nutrition2|Modified
nutrition3|nutrition3|nutrition3|Modified
nutrition4|nutrition4|nutrition4|Modified
nutrition5||nutrition5|Created
thirst5||thirst5|Created
thirst4||thirst4|Created
thirst3||thirst3|Created
thirst2||thirst2|Created
thirst1||thirst1|Created
thirst0||thirst0|Created
</details>
<details><summary>icons/mob/screen/old.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
nutrition0|nutrition0|nutrition0|Modified
nutrition2|nutrition2|nutrition2|Modified
nutrition3|nutrition3|nutrition3|Modified
nutrition4|nutrition4|nutrition4|Modified
nutrition5||nutrition5|Created
thirst5||thirst5|Created
thirst4||thirst4|Created
thirst3||thirst3|Created
thirst2||thirst2|Created
thirst1||thirst1|Created
thirst0||thirst0|Created
</details>
<details><summary>icons/mob/screen/orange.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
nutrition0|nutrition0|nutrition0|Modified
nutrition2|nutrition2|nutrition2|Modified
nutrition3|nutrition3|nutrition3|Modified
nutrition4|nutrition4|nutrition4|Modified
nutrition5||nutrition5|Created
thirst5||thirst5|Created
thirst4||thirst4|Created
thirst3||thirst3|Created
thirst2||thirst2|Created
thirst1||thirst1|Created
thirst0||thirst0|Created
</details>
<details><summary>icons/mob/screen/white.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
nutrition0|nutrition0|nutrition0|Modified
nutrition2|nutrition2|nutrition2|Modified
nutrition3|nutrition3|nutrition3|Modified
nutrition4|nutrition4|nutrition4|Modified
nutrition5||nutrition5|Created
thirst5||thirst5|Created
thirst4||thirst4|Created
thirst3||thirst3|Created
thirst2||thirst2|Created
thirst1||thirst1|Created
thirst0||thirst0|Created
</details>

**Comment by LordFowl** [comment]
Stealing my PR's now, are you?

**Comment by Arrow768** [comment]
You could make the robotic recharges lubricate IPCs aswell

**Comment by BurgerLUA** [comment]
I don't know if robotic chargers having lube dispensers would make sense. I rather make a lube dispensing machine near robotics (public access) instead.

**Comment by Arrow768** [comment]
Having a machine that can dispense space lube in the public is somewhat problematic.
I would either add that to the recharges or add a special machine a IPC climbs into, to get lubricated.
But a device that can dispense space-lube will cause issues.

**Comment by BurgerLUA** [comment]
with every space lube dispense, ipcs get a clown horn

**Comment by DickFreedomJohnson** [comment]
Don't use lube for IPC drinks, the obvious best choice would be coolant :^)

**Comment by BurgerLUA** [comment]
updated OP

**Comment by BurgerLUA** [comment]
!review

**Comment by BurgerLUA** [comment]
The thirst mechanics work and everything however it is likely that feedback is required due to the nature of the change and there are may be some things that need balancing.

**Comment by ToasterStrudes** [comment]
>Lubed up IPCs 
I'm enjoying this PR already

**Comment by Juani2400** [comment]
This PR is cool and all that. But I don't like the fact that you can inject any liquid and be hydrated. Injecting water into the bloodstream will kill you, realistically. I know we have no system reflecting this, but injecting vodka or anything of the sort and getting hydrated is a bit too much, imo.

Otherwise good.

**Comment by BurgerLUA** [comment]
Lube as a thirst mechanic was removed for IPCs.

**Comment by SirCatnip** [comment]
I actually really like the idea of thirst mechanics, but, tacking it onto IPCs just seems like an arbitrary nerf to make playing a synthetic more painful than it already is. Humans need to drink water pretty often, an android in the 25th century shouldn't be less efficient than 20th century machines.

**Comment by BurgerLUA** [comment]
Thirst penalties were nerfed. Just need to fix the travis errors and everything will be fine.

**Comment by BurgerLUA** [comment]
Yeah it's done. Everything is up to snuff according to the poll results.

**Comment by BurgerLUA** [comment]
Screenshots of noticable changes: https://imgur.com/a/FuToQxa

Everything is is small tweaks to various parts of the station, including adding more watertanks to maintenance so antags always have a water source.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 1]
```diff
@@ -222,14 +222,14 @@
...
 	msg += "<span class='warning'>"
-
-
-	if (!(src.species.flags & NO_CHUBBY))
-		if(nutrition < max(max_nutrition - 300, 0))
+	if (!(src.species.flags & NO_CHUBBY) && max_nutrition > 0)
+		if(nutrition / max_nutrition <= CREW_NUTRITION_VERYHUNGRY)
 			msg += "[T.He] [T.is] severely malnourished.\n"
-		else if(nutrition >= max_nutrition + 100)
+		else if(nutrition / max_nutrition >= CREW_NUTRITION_OVEREATEN)
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/examine.dm]: This condenses down to `if (maths) {} else {}`. Doesn't need to be an elseif.

**Pull request merged**